expandybird/expander/expander.go

```diff
@@ -46,6 +46,30 @@ func NewExpander(binary string) Expander {
 	return &expander{binary}
 }
 
+// NewTemplateFromType creates and returns a new template whose content
+// is a YAML marshaled resource assembled from the supplied arguments.
+func NewTemplateFromType(name, typeName string, properties map[string]interface{}) (*common.Template, error) {
```

Comment:
Well, all of the other NewTemplateFrom... methods are in this package, in fact in this file. One could argue that the whole set of methods could go into a package used by clients. I'd buy that argument. However, unless/until we do that, this is right place for all NewTemplateFrom... methods.